src/Missings.jl

```diff
@@ -514,4 +514,88 @@ julia> emptymissing(first)([1], 2)
 """
 emptymissing(f) = (x, args...; kwargs...) -> isempty(x) ? missing : f(x, args...; kwargs...)
 
-end # module
+# Only for internal use. Allows dispatch over anonymous functions.
+struct MissingSmallest{T}
+    lt::T
+end
+
+"""
+    missingsmallest(f)
+
+Return a function of two arguments `x` and `y` that tests whether `x` is less
+than `y` such that `missing` is always less than the other argument. In other 
+words, return a modified version of the partial order function `f` such that
+`missing` is the smallest possible value, and all other non-`missing` values are
+compared according to `f`.
+
+The behavior of the standard `isless` function modified to treat `missing` as
+the smallest value can be obtained by calling the 2-argument `missingsmallest(x,
+y)` function. This is equivalent to `missingsmallest(isless)(x, y)`.
+
+# Examples
+```
+julia> lengthmissing = passmissing(length)
+julia> isshorter = missingsmallest((s1, s2) -> isless(lengthmissing(s1), lengthmissing(s2)))
```

Comment:
```suggestion
julia> isshorter = missingsmallest((s1, s2) -> isless(length(s1), length(s2)))
```
`missingsmallest` will take care of missings, we don't also need to `passmissing`.

Reply:
Ah, the reason is that we have `ismissing(y) ? false : ismissing(x) ? true : ms.lt(x, y)` so if `x` or `y` is `missing` then `length` will not be called. Right?

Reply:
yep!